core/Piranha/Models/Comment.cs

```diff
@@ -68,8 +68,31 @@ public sealed class Comment
         /// <summary>
         /// Comments are approved by default unless you use some kind of comment
         /// validation mechanism.
+        /// Note on obsolete:
+        /// IsApproved=True corresponds to Status=CommentStatus.Approved
+        /// IsApproved=False corresponds to Status=CommentStatus.Rejected
+        /// Submitted comments with config set to NOT approve directly will result in
+        /// Status=CommentStatus.Pending
+        /// In other words: there is no way to deal with pending comments using IsApproved
         /// </summary>
-        public bool IsApproved { get; set; } = true;
+        [Obsolete("IsApproved is obsolete and has been replaced with CommentStatus", true)]
```

Comment:
Shouldn't the second parameter be set to `false` as you've added a fallback implementation? Setting the boolean to `true` will render compilation errors when using the property instead of giving a compiler warning.

Reply:
I was not sure on the decision but noted it (things to revise/left to consider) since it made sense to set it to true.

Reply:
Note that I removed the `IsValid`-property completely from `Data` layer, replaced by the `Status`. I considered to be more clean to replace it "under the hood" rather than having both properties present in DB until the old one is eventually removed.